code/mob/living/critter/flock/flockdrone.dm

```diff
@@ -674,6 +681,28 @@
 				src.icon_state = "drone-d2"
 	return
 
+/mob/living/critter/flock/drone/proc/update_health_icon()
```

Comment:
Should probably move this down to `/mob/living/critter/flock` instead of copy pasting

Reply:
I was uncertain because saying src.update_health_icon() doesn't work in some cases in /mob/living/critter/flock, such as in New() with the new names for Flockdrones and Flockbits going to be requiring their real_name set based on the Flock. Would kind of have some cases where src.update_health_icon() can't be written in some procs for /mob/living/critter/flock and needs to be individually set in the Flockdrone and Flockbit files, but the proc is still written in the /mob/living/critter/flock file.

Reply:
maybe don't have it in the flockdrone/bit, but instead with the other annotation stuff? That way you could just iterate over `flock.units` instead?

Reply:
Okay better question, why are we using `real_name` there anyways and not just ref? Using a (technically) mutable name var as a key is questionable.

Reply:
There's a builtin byond `ref(object)` proc which just returns a guaranteed unique string for any object, I think that's a better way of doing it?

Reply:
Seems to be working now using src, not sure why